### DISCUSSION TITLE
Can ADRs be written in AsciiDoc format?

### DISCUSSION THREAD

**Discussion by synclpz** [structurizr/structurizr.github.io#9] [Q&A] [answered]
### Description

At the moment of issue creation the file https://github.com/structurizr/structurizr.github.io/blob/b59cba34418cb2298ed0e9b8b5e52b3c9de4b0fb/ui/decisions/index.md contains line

```
In Structurizr, each ADR has an ID, title, date and status (e.g. "Proposed", "Accepted", "Superseded", etc), along with unstructured content written using Markdown or AsciiDoc. ADRs can either be created manually, or imported from tools like [adr-tools](https://github.com/npryce/adr-tools).
```
This line should be changed to

```
In Structurizr, each ADR has an ID, title, date and status (e.g. "Proposed", "Accepted", "Superseded", etc), along with unstructured content written using Markdown. ADRs can either be created manually, or imported from tools like [adr-tools](https://github.com/npryce/adr-tools).
```

According to https://github.com/structurizr/structurizr.github.io/blob/main/dsl/adrs.md and source for parser https://github.com/structurizr/import/blob/595fa3dfe26dc38803b1adcb0caf4052762e0fc6/src/main/java/com/structurizr/importer/documentation/AdrToolsDecisionImporter.java the only way is to use .md

### Steps to reproduce

Open doc at https://github.com/structurizr/structurizr.github.io/blob/b59cba34418cb2298ed0e9b8b5e52b3c9de4b0fb/ui/decisions/index.md or https://docs.structurizr.com/ui/decisions/.

### Severity

Minor

### Priority

I have no budget and there's no rush, please fix this for free

**Comment by synclpz** [comment]
BTW adrs doc also has a reference to AsciiDoc support at https://github.com/structurizr/structurizr.github.io/blob/b59cba34418cb2298ed0e9b8b5e52b3c9de4b0fb/dsl/adrs.md#L11C44-L11C44

**Comment by simonbrowndotje** [comment]
AsciiDoc decisions can be rendered by the Structurizr UI:

<img width="1728" alt="Screenshot 2023-12-22 at 23 16 54" src="https://github.com/structurizr/structurizr.github.io/assets/1009874/56061e29-eae6-4ed6-a30a-51fc868fd0f6">
<img width="547" alt="Screenshot 2023-12-22 at 23 17 07" src="https://github.com/structurizr/structurizr.github.io/assets/1009874/bee66f31-75df-4cd9-a37c-19e88655d980">

As it says at [https://docs.structurizr.com/dsl/adrs](https://docs.structurizr.com/dsl/adrs):

> The above behaviour can be customised by specifying the fully qualified class name of your own implementation of DocumentationImporter, which needs to be on the DSL classpath or installed as a JAR file in the plugins directory next to your DSL file.

As far as I'm aware though, nobody has yet written (or released publicly) an importer for AsciiDoc ADRs.

**Reply by synclpz** [reply, marked as answer]
I've just written one, actually extended yours (it now just works for both Markdown & AsciiDoc). I can share at as a PR if you wish to have it in your codebase. For us it works as a `plugin` in workspace now.

**Reply by synclpz** [reply]
@simonbrowndotje do you need a universal (md+asciidoc) ADR importer to be included in your DSL repo? I can make a PR, I got code and tests ready. For me it would be great if it could come with structurizr itself and not as a plugin.
